src/data/bulkOperationsProvider.ts

```diff
@@ -0,0 +1,77 @@
+import type { PineconeConfiguration } from './types';
+import type { HTTPHeaders } from '../pinecone-generated-ts-fetch/db_data';
+import {
+  BulkOperationsApi,
+  Configuration,
+  ConfigurationParameters,
+  X_PINECONE_API_VERSION,
+} from '../pinecone-generated-ts-fetch/db_data';
+import {
+  buildUserAgent,
+  getFetch,
+  normalizeUrl,
+  queryParamsStringify,
+} from '../utils';
+import { IndexHostSingleton } from './indexHostSingleton';
+import { middleware } from '../utils/middleware';
+
+export class BulkOperationsProvider {
```

Comment:
**thought:** This provider is really similar to the existing `DataOperationsProvider`: https://github.com/pinecone-io/pinecone-ts-client/blob/main/src/data/dataOperationsProvider.ts

While they ultimately return different classes, I wonder if we could make this factory a bit more generic and share some of this code. (I saw you just added a ticket for this, nice 👍).

Reply:
Yes, I've got a ticket to address this, totally agree!